### PR TITLE
Use label to describe repository type

### PR DESCRIPTION
**Pull request by CirnoT**
chrome_2020-06-15_14-45-46
chrome_2020-06-15_15-17-28
chrome_2020-06-15_15-17-40
chrome_2020-06-15_15-17-53
chrome_2020-06-15_15-18-18

### CODE REVIEW

**Comment by CirnoT** [comment]
Previous style for reference:
chrome_2020-06-15_10-38-55
chrome_2020-06-15_10-39-06

**Comment by CirnoT** [comment]
Noticed that archived icon would look weird after the label so changed that too.

chrome_2020-06-15_10-59-34
chrome_2020-06-15_10-59-45

**Comment by CirnoT** [comment]
Reworked this now to do it everywhere. This should now follow GH style more closely, which we kind of must do in some places because we're using octicons and have to go around some missing icons.

We should be good to safely use octicons v2 as per https://github.com/go-gitea/gitea/issues/11889 without any ill-effect of losing certain information.

Additionally this PR resolves the issue whereas templates where not marked on explore page at all.

**Comment by CirnoT** [comment]
A slight note is that this does not touch dashboard with repository list so that would be one place where private template is not marked after octicons v2.

The text on dashboard is generated dynamically client-side and we can't use translations there for label, nor it would look very well on such small space.

File: templates/repo/header.tmpl [hunk 1]
```diff
@@ -6,15 +6,51 @@
...
+								{{svg "octicon-repo" 32}}
+							{{end}}
+						{{else if .IsMirror}}
+							{{svg "octicon-repo-clone" 32}}
+						{{else if .IsFork}}
+							{{svg "octicon-repo-forked" 32}}
+						{{else}}
+							{{svg "octicon-repo" 32}}
+						{{end}}
+					{{end}}
```
Comment (silverwind) [inline templates/repo/header.tmpl]: Would've still kept the `repo/header_icon` file. Even if it's not duplicated, it's still nice to have file-level abstraction sometimes.

File: templates/explore/repo_list.tmpl [hunk 2]
```diff
@@ -7,13 +7,26 @@
...
 				</a>
-				{{if .IsPrivate}}
-					<span class="middle text gold">{{svg "octicon-lock" 16}}</span>
-				{{else if and (not .IsMirror) (not .IsFork) (.Owner.Visibility.IsPrivate) }}
-					<span class="text gold">{{svg "octicon-internal-repo" 16}}</span>
-				{{else if .IsFork}}
+				{{if .IsArchived}}<span class="ui compact label">{{$.i18n.Tr "repo.desc.archived"}}</span>{{end}}
+				{{if .IsTemplate}}
+					{{if .IsPrivate}}
+						<span class="ui basic label">{{$.i18n.Tr "repo.desc.private_template"}}</span>
```
Comment (silverwind) [inline templates/explore/repo_list.tmpl:14]: Why not render 2 labels? E.g. `private` `template`?

Comment (CirnoT) [reply templates/explore/repo_list.tmpl:14]: Takes more space and looks weird imho, makes unnecessary distinction with a visible margin and usually user is interested in content of both, not separately.

**Review by silverwind — APPROVED** [review]

**Comment by lunny** [comment]
I think multiple icons or lables are better.

The repositories could be private or not, template or not, archive or not, forked or not.
If a private template repository is a fork and has been archived, it will display 4 icons or lables.

Or keep private and fork as icons but template and archive as lables.

**Comment by CirnoT** [comment]
Current variations should be:

[Private template]
[Template]
[Private]
[Internal]
[Internal template]

[Archived]

(Fork/Mirror icon)

--

They can freely mix and match between these 3 sets, eg.
[Internal template] [Archived] (Fork icon)
or
[Private] (Mirror icon)

**Review by 6543 — APPROVED** [review]

**Comment by techknowlogick** [comment]
ping lg-tm

**Pull request merged**